logstash-core/lib/logstash/runner.rb

```diff
@@ -285,6 +285,12 @@ def execute
     require "logstash/util/java_version"
     require "stud/task"
 
+    if Process.euid() == 0
+      run_as_root = setting("run_as.root") == true
+      raise(RuntimeError,"can not run logstash as root") if !run_as_root
+      logger.warn("running logstash as root is not recommended because of security reasons") if run_as_root
```

Comment:
If we're preventing Logstash from running as root, the error message should reflect that rather than saying it's not recommended. 

Something like, "Logstash cannot be run as root for security reasons

Reply:
This suggestion LGTM!